File: ios/App/App/Services/AudioManager.swift
import Foundation
import Combine
import AVFoundation

public struct SoundTrack: Identifiable, Hashable {
    public let id: String
    public let name: String
    public let icon: String
    public var volume: Float // 0.0 to 1.0
    public var isMuted: Bool

    public init(id: String, name: String, icon: String, volume: Float = 0.0, isMuted: Bool = false) {
        self.id = id
        self.name = name
        self.icon = icon
        self.volume = volume
        self.isMuted = isMuted
    }
}

@MainActor
public final class AudioManager: ObservableObject {
    public static let shared = AudioManager()

    @Published public var isPlaying: Bool = false
    @Published public var masterVolume: Float = 0.8
    @Published public var tracks: [SoundTrack] = [
        SoundTrack(id: "rain", name: "Pluie Douce", icon: "cloud.rain.fill", volume: 0.5),
        SoundTrack(id: "heavy_rain", name: "Pluie Forte", icon: "cloud.heavyrain.fill", volume: 0.0),
        SoundTrack(id: "thunder", name: "Tonnerre", icon: "cloud.bolt.fill", volume: 0.2),
        SoundTrack(id: "forest", name: "Forêt & Oiseaux", icon: "leaf.fill", volume: 0.0),
        SoundTrack(id: "ocean", name: "Vagues de l'Océan", icon: "water.waves", volume: 0.0),
        SoundTrack(id: "fireplace", name: "Feu de Cheminée", icon: "flame.fill", volume: 0.0),
        SoundTrack(id: "cafe", name: "Ambiance Café", icon: "cup.and.saucer.fill", volume: 0.0),
        SoundTrack(id: "night", name: "Nuit d'Été", icon: "moon.stars.fill", volume: 0.0),
        SoundTrack(id: "wind", name: "Brise & Vent", icon: "wind", volume: 0.0),
        SoundTrack(id: "brown_noise", name: "Bruit Brun", icon: "waveform", volume: 0.0)
    ]

    public init() {}

    public func togglePlayback() {
        isPlaying.toggle()
        HapticManager.shared.medium()
    }

    public func setTrackVolume(id: String, volume: Float) {
        if let index = tracks.firstIndex(where: { $0.id == id }) {
            tracks[index].volume = volume
            if volume > 0 && !isPlaying {
                isPlaying = true
            }
        }
    }

    public func muteAll() {
        for index in tracks.indices {
            tracks[index].volume = 0.0
        }
        isPlaying = false
        HapticManager.shared.light()
    }
}

File: android/app/src/main/java/dev/ethone/app/service/AudioManager.kt
package dev.ethone.app.service

import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableFloatStateOf
import androidx.compose.runtime.mutableStateListOf
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue

data class SoundTrack(
    val id: String,
    val name: String,
    val iconName: String,
    var volume: Float = 0f
)

object AudioManager {
    var isPlaying by mutableStateOf(false)
        private set

    var masterVolume by mutableFloatStateOf(0.8f)

    val tracks = mutableStateListOf(
        SoundTrack("rain", "Pluie Douce", "water_drop", 0.5f),
        SoundTrack("heavy_rain", "Pluie Forte", "thunderstorm", 0.0f),
        SoundTrack("thunder", "Tonnerre Lointain", "bolt", 0.2f),
        SoundTrack("forest", "Forêt & Oiseaux", "park", 0.0f),
        SoundTrack("ocean", "Vagues de l'Océan", "waves", 0.0f),
        SoundTrack("fireplace", "Feu de Cheminée", "local_fire_department", 0.0f),
        SoundTrack("cafe", "Ambiance Café", "coffee", 0.0f),
        SoundTrack("night", "Nuit d'Été", "bedtime", 0.0f),
        SoundTrack("wind", "Brise & Vent", "air", 0.0f),
        SoundTrack("brown_noise", "Bruit Brun", "graphic_eq", 0.0f)
    )

    fun togglePlayback() {
        isPlaying = !isPlaying
    }

    fun setTrackVolume(id: String, volume: Float) {
        val index = tracks.indexOfFirst { it.id == id }
        if (index != -1) {
            tracks[index] = tracks[index].copy(volume = volume)
            if (volume > 0f && !isPlaying) {
                isPlaying = true
            }
        }
    }

    fun muteAll() {
        for (i in tracks.indices) {
            tracks[i] = tracks[i].copy(volume = 0f)
        }
        isPlaying = false
    }
}
